### PR TITLE
saving of multiform proportions so they will be preserved when multif…

### PR DESCRIPTION
**Pull request by katush**
saving multiform proportions so they are preserved when the multiform is switched to different vis

### CODE REVIEW

File: src/multiform/MultiForm.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ import {
 import {IMultiForm, IMultiFormOptions, addSelectVisChooser, addIconVisChooser} from './IMultiForm';
 import {createNode, ProxyMetaData, clearNode, selectVis} from './internal';
 import {Range} from '../range';
+import {action} from '../provenance/ActionNode';
```
Comment (thinkh) [inline src/multiform/MultiForm.ts:19]: Is this import necessary?

**Review by thinkh — COMMENTED** [review]
Do we need this import?

**Comment by thinkh** [comment]
@katush Now Travis is not happy. I still don't know where we need this import, but it seems that we need it. ;-) Sorry for the inconvenience. 😃

**Comment by katush** [comment]
No problem, in my local copy it was marked as unused as well.. strange, should be fine now

File: src/multiform/MultiForm.ts [hunk 2]
```diff
@@ -134,13 +135,17 @@ export default class MultiForm extends AVisInstance implements IVisInstance, IMu
         };
         this.actVis.on('transform', t);
         const r = this.actVis.transform(scale, rotate);
+        this.options.all.heightTo = this.actVis.rawSize[1] * (scale[1]||1);
```
Comment (sgratzl) [inline src/multiform/MultiForm.ts]: this sounds like a hack, which may only work for mothertable but not in general. In addition, why is `this.actVis.size[1]` not enough?

File: src/multiform/MultiForm.ts (hunk 1 above)
Comment (sgratzl) [reply src/multiform/MultiForm.ts:19]: what was the travis error after removing it?

Comment (katush) [reply src/multiform/MultiForm.ts:19]: The Job Log was empty if I remember correctly

File: src/multiform/MultiForm.ts (hunk 2 above)
Comment (katush) [reply src/multiform/MultiForm.ts]: With the edited visualizations it should work everywhere, it is not dependent on mother table code in any way. Though you are right  that `this.actVis.size[1]` should be enough.

File: src/multiform/MultiForm.ts (hunk 1 above)
Comment (sgratzl) [reply src/multiform/MultiForm.ts:19]: travis is having problems at to moment fetching job logs

**Comment by sgratzl** [comment]
I still vote that this is either an optional behavior or a multiform provides a hook before switching visualizations to accomplish a similar thing.

**Comment by katush** [comment]
@sgratzl The only other way is to somehow recalculate and set the `scale` option. But in many visualizations, the `rawSize` depends of the size of the data and thus you cannot calculate the scale before initializing the vis.
This led me to introducing the `heightTo` and `width` options which enable calculating initial scale "on the fly" when initializing visualizations. And then it is easier to pass this attributes rather than the scale.
But if you have more efficient solution, pleas explain, I would be glad to learn.

**Comment by thinkh** [comment]
I created a `mothertable` branch for this repository and merge your PR there.

**Pull request merged**